includes/multi-currency/class-currency-switcher-widget.php

```diff
@@ -0,0 +1,172 @@
+	 * @param array $instance Saved values from database.
+	 */
+	public function widget( $args, $instance ) {
+		$instance = wp_parse_args(
+			$instance,
+			self::DEFAULT_SETTINGS
+		);
+
+		$title = apply_filters( 'widget_title', $instance['title'] );
+
+		echo $args['before_widget']; // phpcs:ignore WordPress.Security.EscapeOutput
+		if ( ! empty( $title ) ) {
+			echo $args['before_title'] . $title . $args['after_title']; // phpcs:ignore WordPress.Security.EscapeOutput
+		}
+
+		?>
+		<form>
+			<select
+				name="currency"
+				aria-label="<?php echo esc_attr( $title ); ?>"
+				onchange="this.form.submit()"
+			>
+				<?php
+				foreach ( $this->multi_currency->get_enabled_currencies() as $currency ) {
+					$this->display_currency_option( $currency, $instance['symbol'], $instance['flag'] );
+				}
+				?>
+			</select>
+		</form>
+		<?php
+
+		echo $args['after_widget']; // phpcs:ignore WordPress.Security.EscapeOutput
+	}
+
+	/**
+	 * Back-end widget form.
+	 *
+	 * @param array $instance Previously saved values from database.
+	 */
+	public function form( $instance ) {
+		$instance = wp_parse_args(
+			$instance,
+			self::DEFAULT_SETTINGS
+		);
+		?>
+		<p>
+			<label for="<?php echo esc_attr( $this->get_field_name( 'title' ) ); ?>">
+				<?php esc_html_e( 'Title:', 'woocommerce-payments' ); ?>
+			</label>
+			<input
+				class="widefat"
+				id="<?php echo esc_attr( $this->get_field_id( 'title' ) ); ?>"
+				name="<?php echo esc_attr( $this->get_field_name( 'title' ) ); ?>"
+				type="text"
+				value="<?php echo esc_attr( $instance['title'] ); ?>"
+			/>
+		</p>
+		<p>
+			<input
+				class="checkbox"
+				id="<?php echo esc_attr( $this->get_field_id( 'symbol' ) ); ?>"
+				name="<?php echo esc_attr( $this->get_field_name( 'symbol' ) ); ?>"
+				type="checkbox"<?php checked( $instance['symbol'] ); ?>
+			/>
+			<label for="<?php echo esc_attr( $this->get_field_id( 'symbol' ) ); ?>">
+				<?php esc_html_e( 'Display curency symbols', 'woocommerce-payments' ); ?>
+			</label>
+			<br/>
+			<input
+				class="checkbox"
+				id="<?php echo esc_attr( $this->get_field_id( 'flag' ) ); ?>"
+				name="<?php echo esc_attr( $this->get_field_name( 'flag' ) ); ?>"
+				type="checkbox"<?php checked( $instance['flag'] ); ?>
+			/>
+			<label for="<?php echo esc_attr( $this->get_field_id( 'flag' ) ); ?>">
+				<?php esc_html_e( 'Display flags', 'woocommerce-payments' ); ?>
+			</label>
+		</p>
+		<?php
+	}
+
+	/**
+	 * Sanitize widget form values as they are saved.
+	 *
+	 * @param array $new_instance Values just sent to be saved.
+	 * @param array $old_instance Previously saved values from database.
+	 *
+	 * @return array Updated safe values to be saved.
+	 */
+	public function update( $new_instance, $old_instance ) {
+		$instance = [
+			'title'  => sanitize_text_field( $new_instance['title'] ),
+			'symbol' => ( $new_instance['symbol'] ?? null ) ? 1 : 0,
+			'flag'   => ( $new_instance['flag'] ?? null ) ? 1 : 0,
```

Comment:
The null coalescing operator only returns the second operand if the first is null. Is it needed here, or can we use `$new_instance['symbol'] ? 1 : 0`?

Reply:
I didn't come up with something more appealing. The value received from the checkbox is a string with 'on' if is checked and the key is missing from the array when is unchecked. So I thought this is maybe the shortest way to write it.

Using only the coalescing will result in `'on'` or `0`.
And using `$new_instance['symbol'] ? 1 : 0` like before doesn't show any error on the admin, but produces a fatal error on tests because the key is missing from the array.

Reply:
Gotcha! I didn't know that the key would not be defined when unchecked. Thanks for clarifying!

IMO, `isset( $new_instance['symbol'] ) ? 1 : 0` would be clearer, but both should work fine, so I'll defer this to you 🙂

Reply:
Sometimes I think too much on using new syntax. 💯 more readable with isset 🙂  Addressed in 6dbe5f9